frontend/src/metabase/query_builder/components/notebook/ReadOnlyNotebook.unit.spec.tsx

```diff
@@ -0,0 +1,185 @@
+import React from "react";
+
+import { renderWithProviders, screen, getIcon } from "__support__/ui";
+import { createEntitiesState } from "__support__/store";
+import { setupDatabasesEndpoints } from "__support__/server-mocks/database";
+import { setupSearchEndpoints } from "__support__/server-mocks/search";
+
+import type { DatasetQuery } from "metabase-types/types/Card";
+
+import { createMockState } from "metabase-types/store/mocks";
+import { createMockStructuredDatasetQuery } from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+
+import { getMetadata } from "metabase/selectors/metadata";
+
+import Question from "metabase-lib/Question";
+
+import ReadOnlyNotebook from "./ReadOnlyNotebook";
+
+const makeQuery = (options: any) => {
+  // we have to cast this because we have 2 incompatible DatasetQuery types
+  return createMockStructuredDatasetQuery({
+    query: options,
+  }) as DatasetQuery;
+};
```

Comment:
this is kinda nasty, but the StructuredQuery type in our api types is missing most of its properties. Adding them, however, throws tons of type errors throughout the app 😬  Definitely worth fixing, but I didn't want to blow out the scope of this PR by opening that can of worms.